openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIComponentMapper.java

```diff
@@ -217,6 +210,61 @@ private ObjectSchema generateObjectSchemaFromRecordFields(Map<String, Schema> sc
         return componentSchema;
     }
 
+    /**
+     * This function uses to handle the field datatype has TypeReference(ex: Record or Enum).
+     */
+    private Schema handleTypeReference(Map<String, Schema> schema, TypeSymbol typeReferenceSymbol,
+                                       Schema property) {
+        if (((TypeReferenceTypeSymbol) typeReferenceSymbol).definition().kind() == SymbolKind.ENUM) {
+            TypeReferenceTypeSymbol typeRefEnum = (TypeReferenceTypeSymbol) typeReferenceSymbol;
+            EnumSymbol enumSymbol = (EnumSymbol) typeRefEnum.definition();
+            property = mapEnumValues(enumSymbol);
+        } else {
+            property.set$ref(typeReferenceSymbol.getName().orElseThrow().trim());
+            TypeSymbol recordVariable = typeReferenceSymbol;
+            TypeReferenceTypeSymbol typeRecord = (TypeReferenceTypeSymbol) recordVariable;
+            createComponentSchema(schema, typeRecord);
+        }
+        return property;
+    }
```

Comment:
Shall we change the method to accept an instance of `TypeReferenceTypeSymbol`, to avoid all the redundant type castings?

Reply:
Fix  91753a9